path/normalize.ts

```diff
@@ -26,6 +26,34 @@ import { normalize as windowsNormalize } from "./windows/normalize.ts";
  * @param path Path to be normalized
  * @returns The normalized path.
  */
-export function normalize(path: string): string {
+export function normalize(path: string): string;
+/**
+ * Normalize the path, resolving `'..'` and `'.'` segments.
+ *
+ * @experimental **UNSTABLE**: New API, yet to be vetted.
+ *
+ * Note: Resolving these segments does not necessarily mean that all will be
+ * eliminated. A `'..'` at the top-level will be preserved, and an empty path is
+ * canonically `'.'`.
+ *
+ * @example Usage
+ * ```ts
+ * import { normalize } from "@std/path/normalize";
+ * import { assertEquals } from "@std/assert";
+ *
+ * if (Deno.build.os === "windows") {
+ *   assertEquals(normalize("C:\\foo\\bar\\..\\baz\\quux"), "C:\\foo\\baz\\quux");
+ *   assertEquals(normalize(new URL("file:///C:/foo/bar/../baz/quux")), "C:\\foo\\baz\\quux");
+ * } else {
+ *   assertEquals(normalize("/foo/bar/../baz/quux"), "/foo/baz/quux");
+ *   assertEquals(normalize(new URL("file:///foo/bar/../baz/quux")), "/foo/baz/quux");
+ * }
+ * ```
+ *
+ * @param path Path to be normalized. Path can be a string or a file URL object.
+ * @returns The normalized path.
+ */
+export function normalize(path: string | URL): string;
```

Comment:
```suggestion
export function normalize(path: string): string;
```

Reply:
That is the same as the 1st overload

Reply:
Sorry, I mean `URL`. Ditto elsewhere.

Reply:
I think `string | URL` is better for this type. When we stabilize URL input, this is going to be the only one signature of this function as is. This prepares for it better.

Also we don't need to use `as any` in `path/normalize.ts` by using this type.

Reply:
That's not how we did the other PRs.

Reply:
I'd suggest we should fix others

Reply:
Ok. Are you able to fix the others?

Reply:
Sure